Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Upsert.ts

```diff
@@ -125,6 +144,13 @@ export async function upsertDocument(
   } catch (e) {
     Logger.error(`${moduleName}.upsertDocument`, traceId, e);
 
+    if (e.message === '[MongoServerError: WriteConflict]') {
```

Comment:
This is good, but are there any other indications in the error object we can use, apart from a string compare? For example, what if the message changes slightly in a new driver version?

Can you check the type of `e`, maybe it's a `MongoServerError` and there are additional fields on it besides `message`. You may need to look at the TypeScript type definitions to see, or trigger the scenario and look at the exception in debug.

Reply:
Yes, that is a good point. Thanks.